forui/lib/src/widgets/slider/slider.dart

```diff
@@ -0,0 +1,342 @@
+  ///
+  /// In practice, this is mostly useful for range sliders.
+  // ignore: avoid_positional_boolean_parameters
+  final String Function(double) semanticValueFormatterCallback;
+
+  /// Creates a [FSlider].
+  FSlider({
+    required this.controller,
+    this.style,
+    this.layout = Layout.ltr,
+    this.marks = const [],
+    this.enabled = true,
+    this.tooltipBuilder = _tooltipBuilder,
+    this.semanticValueFormatterCallback = _semanticValueFormatter,
+    String Function(FSliderSelection)? semanticFormatterCallback,
+    super.key,
+  }) : semanticFormatterCallback = semanticFormatterCallback ?? _formatter(controller);
+
+  @override
+  State<FSlider> createState() => _FSliderState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('controller', controller))
+      ..add(DiagnosticsProperty('style', style))
+      ..add(EnumProperty('layout', layout))
+      ..add(IterableProperty('marks', marks))
+      ..add(FlagProperty('enabled', value: enabled, ifTrue: 'enabled', ifFalse: 'disabled'))
+      ..add(ObjectFlagProperty.has('tooltipBuilder', tooltipBuilder))
+      ..add(ObjectFlagProperty.has('semanticFormatterCallback', semanticFormatterCallback))
+      ..add(ObjectFlagProperty.has('semanticValueFormatterCallback', semanticValueFormatterCallback));
+  }
+}
+
+class _FSliderState extends State<FSlider> {
+  @override
+  Widget build(BuildContext context) {
+    final styles = context.theme.sliderStyles;
+    final style = widget.style ??
+        switch ((widget.enabled, widget.layout.vertical)) {
+          (true, false) => styles.enabledHorizontalStyle,
+          (true, true) => styles.enabledVerticalStyle,
+          (false, false) => styles.disabledHorizontalStyle,
+          (false, true) => styles.disabledVerticalStyle,
+        };
+
+    return InheritedData(
+      style: style,
+      layout: widget.layout,
+      marks: widget.marks,
+      enabled: widget.enabled,
+      tooltipBuilder: widget.tooltipBuilder,
+      semanticFormatterCallback: widget.semanticFormatterCallback,
+      semanticValueFormatterCallback: widget.semanticValueFormatterCallback,
+      child: LayoutBuilder(
+        builder: (context, constraints) => SliderLayout(
+          controller: widget.controller,
+          style: style,
+          layout: widget.layout,
+          marks: widget.marks,
+          constraints: constraints,
+        ),
+      ),
+    );
+  }
+}
+
+/// A slider's styles.
+final class FSliderStyles with Diagnosticable {
+  /// The enabled slider's horizontal style.
+  final FSliderStyle enabledHorizontalStyle;
+
+  /// The enabled slider's vertical style.
+  final FSliderStyle enabledVerticalStyle;
+
+  /// The disabled slider's horizontal style.
+  final FSliderStyle disabledHorizontalStyle;
+
+  /// The disabled slider's vertical style.
+  final FSliderStyle disabledVerticalStyle;
```

Comment:
Should we include an error state?

Reply:
I'm open to it but what how should the slider in an error state look like?